addons/hr_work_entry_contract/models/hr_contract.py

```diff
@@ -100,6 +101,35 @@ def _get_attendance_intervals(self, start_dt, end_dt):
             ))
         return result
 
+    def _postprocess_attendance_intervals(self, intervals):
+        # _attendance_intervals_batch combines the attendances regardless of work entry type if their date and time overlap.
+        # This makes a single attendance block with the work entry type of the first attendance only. This function undoes those
+        # undesirable merges
+        # intervals: [(start_date, end_date, attendances)]
+        try:
+            multi_type_intervals = [interval for interval in intervals._items if len(interval[2].mapped('work_entry_type_id')) > 1]
+        except KeyError:
+            return intervals
+        for interval in multi_type_intervals:
+            attendances = interval[2]
+            current_work_entry_type = attendances[0].work_entry_type_id
+            attendances_of_type = []
+            start_date = interval[0]
+            for attendance in attendances:
+                if not attendance.work_entry_type_id or attendance.work_entry_type_id == current_work_entry_type:
```

Comment:
so we mix entry type with no entry type? could be ok but seems a little that eg. no work time entry would be associated to overtime

Reply:
That's a good point, I'll keep that entries without a type separate